sympy/solvers/solveset.py

```diff
@@ -1530,8 +1530,14 @@ def _solve_exponential(lhs, rhs, symbol, domain):
     a_term = a.as_independent(symbol)[1]
     b_term = b.as_independent(symbol)[1]
 
-    a_base, a_exp = a_term.base, a_term.exp
-    b_base, b_exp = b_term.base, b_term.exp
+    if a_term.is_Mul:
+        a_base, a_exp = (a.atoms(Pow)).pop().args
+    else:
+        a_base, a_exp = a_term.base, a_term.exp
```

Comment:
Can you explain in words what this is doing? I'm sure that this is not correct in general.

Should this just be `a_term.as_base_exp()`?